### PR TITLE
Change/selinux detection

### PR DESCRIPTION
**Pull request by stejoo**
SELinux can be tested by Ansible itself instead of relying on an external role dependency. 
This behaviour should be consistent around Ansible 2.4.1. 

I'm calling the fact it in the namespaced way, which got introduced in Ansible 2.5, so I upped the minimal version required to 2.5.

### CODE REVIEW

**Comment by manics** [comment]
@stejoo Thanks for your contribution! The `selinux_utils` role also ensures some dependencies required by the Ansible selinux modules are installed: https://github.com/ome/ansible-role-selinux-utils/blob/2.0.1/tasks/main.yml#L28-L36

For example see the seboolean module docs https://docs.ansible.com/ansible/2.6/modules/seboolean_module.html

Unfortunately we can't test selinux in Docker (it'll have to be done with Vagrant or some other driver). In the past we've had VM images that have selinux enabled but don't include the related Python libraries.

**Comment by stejoo** [comment]
Ah yes the Python requirements. The Ansible check also requires `libselinux-python` so that should be present to even hit it, but it might still go wrong when `libsemanage-python` is not there. That would make it able to detect SELinux status, but unable to run the `seboolean` module ans might cause an issue.

Would it be acceptable if I put back the dependency on the `selinux_role`, while leaving the new `when:` condition in place?

**Comment by manics** [comment]
Hi @stejoo
Sorry for droping this, we caught it whilst cleaning up some issues across our repositaries. Is this still something you require?

**Comment by stejoo** [comment]
Require... that would be too strong. I simply modified this to use an Ansible fact instead of a variable from another role because that's possible since Ansible 2.5 and I think it's cleaner when you can drop a dependency like that.

I forked this role and modified your code here, so I felt I should contribute my modification back to you.
I did make other extensions on my role, various fixes and added the access control bits of Serhii Vasyliev's fork of this role. Feel free to checkout branches within my fork. The 'stejoo' branch has them merged in, the others are sort of feature/fix branches that could be merged here if you find them interesting.

TL;DR: It's up to you whether you want to accept this change or not. I don't mind. Recommend you check out branches of my fork for features/fixes that might interest you.

**Comment by manics** [comment]
Thanks for pointing us to your other features. I've added a [note to the readme](https://github.com/ome/ansible-role-rsync-server#related-roles) about your fork- we have a fairly minimal rsync config for public downloads only, so you're probably in a better position to maintain the other features.
